toolchain/check/type_structure.cpp

```diff
@@ -143,266 +140,108 @@ class TypeStructureBuilder {
-          AppendStructuralConcrete(type_id);
-          break;
-        }
-        case CARBON_KIND(SemIR::IntType int_type): {
-          if (type_id.is_concrete()) {
-            AppendStructuralConcrete(type_id);
-          } else {
-            AppendStructuralConcreteOpenParen(type_id);
-            Push(CloseType());
-            PushArgs({int_type.bit_width_id});
-          }
-          break;
-        }
-
-          // ==== Aggregate types ====
-
-        case CARBON_KIND(SemIR::ArrayType array_type): {
-          AppendStructuralConcreteOpenParen(TypeStructure::ConcreteNoneType());
-          Push(CloseType());
-          PushInstId(array_type.element_type_inst_id);
-          PushInstId(array_type.bound_id);
-          break;
-        }
-        case CARBON_KIND(SemIR::ClassType class_type): {
-          auto args = GetSpecificArgs(class_type.specific_id);
-          if (args.empty()) {
-            AppendStructuralConcrete(class_type.class_id);
-          } else {
-            AppendStructuralConcreteOpenParen(type_id);
-            Push(CloseType());
-            PushArgs(args);
-          }
-          break;
-        }
-        case CARBON_KIND(SemIR::ConstType const_type): {
-          // We don't put the `const` into the type structure since it is a
-          // modifier; just move to the inner type.
-          PushInstId(const_type.inner_id);
-          break;
-        }
-        case CARBON_KIND(SemIR::ImplWitnessAssociatedConstant assoc): {
-          Push(assoc.type_id);
-          break;
-        }
-        case CARBON_KIND(SemIR::PointerType pointer_type): {
-          AppendStructuralConcreteOpenParen(TypeStructure::ConcreteNoneType());
-          Push(CloseType());
-          PushInstId(pointer_type.pointee_id);
-          break;
-        }
-        case CARBON_KIND(SemIR::TupleType tuple_type): {
-          auto inner_types =
-              context_->inst_blocks().Get(tuple_type.type_elements_id);
-          if (inner_types.empty()) {
-            AppendStructuralConcrete(type_id);
-          } else {
-            AppendStructuralConcreteOpenParen(
-                TypeStructure::ConcreteNoneType());
-            Push(CloseType());
-            PushArgs(context_->inst_blocks().Get(tuple_type.type_elements_id));
-          }
-          break;
-        }
-        case CARBON_KIND(SemIR::StructType struct_type): {
-          auto fields =
-              context_->struct_type_fields().Get(struct_type.fields_id);
-          if (fields.empty()) {
-            AppendStructuralConcrete(type_id);
-          } else {
-            AppendStructuralConcreteOpenParen(type_id);
-            Push(CloseType());
-            for (const auto& field : llvm::reverse(fields)) {
-              PushInstId(field.type_inst_id);
-            }
-          }
-          break;
-        }
-        default:
-          CARBON_FATAL("Unhandled type instruction {0}", inst_id);
+  auto BuildTypeStructure(SemIR::TypeIterator type_iter) -> void {
```

Comment:
Perhaps move this out-of-line so that it's easy to read the API of `TypeStructureBuilder`?